stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/nativeimpl/channel/BasicPublish.java

```diff
@@ -56,12 +58,12 @@ public void execute(Context context) {
         BMap<String, BValue> channelObject = (BMap<String, BValue>) context.getRefArgument(0);
         Channel channel = RabbitMQUtils.getNativeObject(channelObject, RabbitMQConstants.CHANNEL_NATIVE_OBJECT,
                 Channel.class, context);
-        String message = context.getStringArgument(0);
-        String routingKey = context.getStringArgument(1);
-        // Handle other message types
-        String exchange = context.getStringArgument(2);
+        BValue msgContent = context.getRefArgument(1);
+        String routingKey = context.getStringArgument(0);
+        String exchange = context.getStringArgument(1);
         try {
-            ChannelUtils.basicPublish(channel, routingKey, message, exchange);
+            ChannelUtils.basicPublish(channel, routingKey, msgContent.stringValue().getBytes(Charset.forName("UTF-8")),
```

Comment:
You can use the constant for "UTF-8" from StandardCharsets class[1]

[1] https://docs.oracle.com/javase/8/docs/api/index.html?java/nio/charset/StandardCharsets.html

Reply:
Fixed